applications/SystemIdentificationApplication/custom_sensors/measurement_residual_response_function.cpp

```diff
@@ -146,15 +145,19 @@ double MeasurementResidualResponseFunction::CalculateValue(ModelPart& rModelPart
 {
     KRATOS_TRY
 
+    double sum_B_p = 0.0;
     double value = 0.0;
     for (auto& p_sensor : mpSensorsList) {
         const double sensor_value = p_sensor->CalculateValue(rModelPart);
         p_sensor->SetSensorValue(sensor_value);
-        const double current_sensor_error_square  =  std::pow(sensor_value - p_sensor->GetValue(SENSOR_MEASURED_VALUE), 2) * 0.5;
-        p_sensor->SetValue(SENSOR_ERROR, current_sensor_error_square);
-        value += std::pow(p_sensor->GetWeight() * current_sensor_error_square, mPCoefficient);
+        const double current_sensor_error = sensor_value - p_sensor->GetValue(SENSOR_MEASURED_VALUE);
+        p_sensor->SetValue(SENSOR_ERROR, current_sensor_error);
+        sum_B_p += ( std::pow( 0.5 * pow(current_sensor_error, 2) * p_sensor->GetWeight(), mPCoefficient ) );
+        value += p_sensor->GetWeight() * std::pow(sensor_value - p_sensor->GetValue(SENSOR_MEASURED_VALUE), 2) * 0.5;
```

Comment:
why do you need the `value` here? is it used anywhere?

Reply:
testing p=1 and old code

Reply:
removed